src/mscorlib/shared/System/Diagnostics/StackFrame.cs

```diff
@@ -0,0 +1,272 @@
+        {
+            iLineNumber = iLine;
+        }
+
+        internal virtual void SetColumnNumber(int iCol)
+        {
+            iColumnNumber = iCol;
+        }
+
+        internal virtual void SetIsLastFrameFromForeignExceptionStackTrace(bool fIsLastFrame)
+        {
+            fIsLastFrameFromForeignExceptionStackTrace = fIsLastFrame;
+        }
+
+        internal virtual bool GetIsLastFrameFromForeignExceptionStackTrace()
+        {
+            return fIsLastFrameFromForeignExceptionStackTrace;
+        }
+
+        /// <summary>
+        /// Returns the method the frame is executing
+        /// </summary>
+        public virtual MethodBase GetMethod()
+        {
+            Contract.Ensures(Contract.Result<MethodBase>() != null);
+
+            return method;
+        }
+
+        /// <summary>
+        /// Returns the offset from the start of the native (jitted) code for the
+        /// method being executed
+        /// </summary>
+        public virtual int GetNativeOffset()
+        {
+            return offset;
+        }
+
+
+        /// <summary>
+        /// Returns the offset from the start of the IL code for the
+        /// method being executed.  This offset may be approximate depending
+        /// on whether the jitter is generating debuggable code or not.
+        /// </summary>
+        public virtual int GetILOffset()
+        {
+            return ILOffset;
+        }
+
+        /// <summary>
+        /// Returns the file name containing the code being executed.  This
+        /// information is normally extracted from the debugging symbols
+        /// for the executable.
+        /// </summary>
+        public virtual String GetFileName()
+        {
+            return strFileName;
+        }
+
+        /// <summary>
+        /// Returns the line number in the file containing the code being executed.
+        /// This information is normally extracted from the debugging symbols
+        /// for the executable.
+        /// </summary>
+        public virtual int GetFileLineNumber()
+        {
+            return iLineNumber;
+        }
+
+        /// <summary>
+        /// Returns the column number in the line containing the code being executed.
+        /// This information is normally extracted from the debugging symbols
+        /// for the executable.
+        /// </summary>
+        public virtual int GetFileColumnNumber()
+        {
+            return iColumnNumber;
+        }
+
+        /// <summary>
+        /// Format method name assuming the MethodBase information is available.
+        /// </summary>
+        private void FormatMethodNameUsingMethodBase(StringBuilder sb)
+        {
+            Debug.Assert(method != null);
+            
+            sb.Append(method.Name);
+
+            /// <summary>
+            /// deal with the generic portion of the method
+            /// </summary>
+            if (method is MethodInfo && ((MethodInfo)method).IsGenericMethod)
```

Comment:
Also, the use of IsGenericMethod is a code smell - usually one means IsGenericMethodDefinition or IsConstructedGenericMethod.